test/distributed/org/apache/cassandra/distributed/upgrade/MixedModeReadRepairDeleteTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.upgrade;
+
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.api.ConsistencyLevel;
+
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.distributed.shared.AssertUtils.row;
+
+public class MixedModeReadRepairDeleteTest extends UpgradeTestBase
```

Comment:
We could add a comment about why we have split the tests. That could be a hint for someone trying to unify them, or extending any of them with more cases.
```suggestion
/**
 * Test read repair after partial deletions when the cluster nodes are in different versions.
 *
 * This test and {@link MixedModeReadRepairWriteTest} are separated to avoid OOM errors on CI (see CASSANDRA-16237).
 */
public class MixedModeReadRepairDeleteTest extends UpgradeTestBase
```